pkg/cmd/pulumi/project/project_ls.go

```diff
@@ -0,0 +1,216 @@
+			displayOpts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+
+			// Get the current workspace and project if any.
+			ws := pkgWorkspace.Instance
+			var err error
+			var currentProject *workspace.Project
+			currentProject, _, err = ws.ReadProject()
+			if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+				return err
+			}
+
+			// Get the current backend.
+			b, err := cmdbackend.CurrentBackend(ctx, ws, cmdbackend.DefaultLoginManager, currentProject, displayOpts)
+			if err != nil {
+				return err
+			}
+
+			// Handle all backend types uniformly
+			// If user specifically requested organization info but we can't provide it
+			if orgName != "" && !b.SupportsOrganizations() {
+				return fmt.Errorf("organizations are not supported by the current backend (%s)", b.Name())
+			}
+
+			// Create a filter for ListStacks
+			filter := backend.ListStacksFilter{}
+			if orgName != "" {
+				filter.Organization = &orgName
+			}
+
+			// List all stacks and group them by project
+			stackSummaries, _, err := b.ListStacks(ctx, filter, nil)
+			if err != nil {
+				return err
+			}
+
+			// Map to track projects and stack counts
+			projectMap := make(map[string]projectListResult)
+
+			// Group stacks by project
+			for _, stack := range stackSummaries {
+				// Get project name from stack reference
+				projectName, ok := stack.Name().Project()
+				if !ok {
+					// Skip stacks without a project
+					continue
+				}
+
+				// Use the string representation of projectName
+				projectNameStr := string(projectName)
+
+				// For backends that support organizations, get the org from the stack name
+				var org string
+				if b.SupportsOrganizations() {
+					stackFullName := stack.Name().String()
+					org = getOrgFromStackName(stackFullName)
+
+					// Skip if we're filtering by organization and this stack doesn't match
+					if orgName != "" && org != orgName {
+						continue
+					}
+				}
+
+				// Update project entry
+				project, exists := projectMap[projectNameStr]
+				if !exists {
+					project = projectListResult{
+						Name:         projectNameStr,
+						Organization: org,
+						StackCount:   0,
+						Backend:      b.Name(),
+					}
+				}
+				project.StackCount++
+				projectMap[projectNameStr] = project
+			}
+
+			// Convert map to slice for output
+			var results []projectListResult
+			for _, project := range projectMap {
+				results = append(results, project)
+			}
+
+			// If no projects, display a message
+			if len(results) == 0 {
+				if orgName != "" {
+					fmt.Println("No projects found in organization", orgName)
+				} else {
+					fmt.Println("No projects found")
+				}
+				return nil
+			}
+
+			// Output the results
+			if jsonOut {
+				out, err := json.MarshalIndent(results, "", "    ")
+				if err != nil {
+					return err
+				}
+				fmt.Println(string(out))
+			} else {
+				// Display header based on backend type
+				if b.SupportsOrganizations() {
```

Comment:
I don't think we need to display differently based on sass vs diy, both of these can just list each stack with org names if they have them.